### PR TITLE
Adventure Pointers

### PR DESCRIPTION
**Pull request by kashike**
https://github.com/KyoriPowered/adventure/pull/352

### CODE REVIEW

**Review by Proximyst — COMMENTED** [review]
Looks good so far, excited for how this'll turn out

File: Spigot-Server-Patches/0010-Adventure.patch [hunk 1]
```diff
@@ -2324,6 +2332,25 @@ index b51a874e4665f977a154792e6216e03e04525f39..6ab14bccb1fcd108931bf7ec331e60f6
...
++    public <T> java.util.Optional<T> get(final net.kyori.adventure.pointer.Pointer<T> pointer) {
++        if (pointer == net.kyori.adventure.identity.Identity.NAME) {
++            return java.util.Optional.of((T) this.getProfile().getName());
++        } else if (pointer == net.kyori.adventure.identity.Identity.DISPLAY_NAME) {
++            return java.util.Optional.of((T) this.displayName());
++        } else if (pointer == net.kyori.adventure.identity.Identity.UUID) {
++            return java.util.Optional.of((T) this.getUniqueId());
++        } else if (pointer == net.kyori.adventure.permission.PermissionChecker.POINTER) {
++            if (this.adventure$permissions == null) {
++                // todo: allow properly returning NOT_SET?
```
Comment (kashike) [inline Spigot-Server-Patches/0010-Adventure.patch:2345]: Any suggestions on how to handle this?

Comment (zml2008) [reply Spigot-Server-Patches/0010-Adventure.patch:2345]: There's a superperms method for `isPermissionSet`, you could use that?

**Comment by zml2008** [comment]
still relevant

**Comment by kezz** [comment]
Should probably be updated to the new `Pointers` system. Also could add pointers for world/entity names/uuids. Also, teams can have names/display names as well.

**Pull request closed without merging**